operator/internal/common/labels/labels_test.go

```diff
@@ -0,0 +1,45 @@
+package labels
+
+import (
+	"reflect"
+	"testing"
+)
+
+func TestMergeLabels(t *testing.T) {
+	cases := []struct {
+		name      string
+		current   map[string]string
+		newLabels map[string]string
+		expected  map[string]string
+		updated   bool
+	}{
+		{
```

Comment:
Can you cover the following test cases 
- current is nil `current: nil`
- newLabels is nil `newLabels: nil`
- current is empty `current: map[string]string{}`
- newLabels is empty `newLabels: map[string]string{}`
- both current and newLabels is nil `current: nil` and `newLabels: nil`
- current contains keys that are not in newLabels `current: map[string]string{"foo":"bar"}` and `newLabels: map[string]string{"bar": "qux"}`
- current contains both a) keys that are in newLabels and b) keys that are not in newLabels `current: map[string]string{"foo":"bar", "bar":"qux"}` and `newLabels: map[string]string{"bar": "snipper"}